qmd/wide-names.qmd

```diff
@@ -0,0 +1,93 @@
+---
+project:
+  execute-dir: ../
+---
+```{r setup, echo=FALSE}
+knitr::opts_knit$set(root.dir = normalizePath("../"))
+```
+# Long-format, Wide-format, and Wide-names {.unnumbered}
+
+## What are wide-format and long-format data?
+
+Speaking very generally, when recording and storing data it is possible to store data in a wide-format, a long-format, or most oftern some combination of the two. When we talk about a long-format, we're talking about there being one row for each observation, measure, or data point. Conversely, when we talk about wide-format, we're talking about all the observations, measures, or data points of one day being recorded in a row. 
+
+For example, a long-format table for recoding wastewater surveillance data may look something like this:
+
+| date | measure | value | unit |
+|------|---------|-------|------|
+|dec 17| covid N1| 32.1  | gc/mL|
+|dec 17| PMMoV   | 16.2  | gc/mL|
+|dec 17| pH      | 8.2   | unitless|
+|dec 20| covid N1| 34.6  | gc/mL|
+|dec 20| PMMoV   | 19.6  | gc/mL|
+|dec 20| pH      | 7.9   | unitless|
+
+While a wide-format version of that same table above might instead look something like this:
+
+| date | covid N1 (gc/mL) | PMMoV (gc/mL) | pH |
+|------|------------------|---------------|----|
+|dec 17|  32.1            | 16.2          | 8.2|
+|dec 20|  34.6            | 19.6          | 7.9|
+
+You can see that the more measures or kinds of data points we wish to record, the wider the wide-format table gets, and the longer the long-format table gets. 
+
+## Issues with moving between formats
+
+There are various pros and cons to each format, but generally it is ideal for data to be stored and analyzed in a long-format table, while it can be easier to record the data in a wide-format table. One trouble that can arise with this situation is that when transforming data from one format to another, there may be differences in how naming conventions work across formats. This can lead to mistakes in how the data is mapped, and ultimately to mistakes in the database where information is being stored.
+
+## PHES-ODM Approach to wide- and long-formats
+
+The PHES-ODM is designed as a series of long-format tables, but it is our aim to be as robust and easy to use as possible. To avoid issues of incongruity when moving between wide-format data recording templates and the final long-format data storage tables, we have designed a specific series of formulae to generate wide-format column names (or wide-names) using the part IDs for long-format tables, headers, and values. This may, wide-names can be parsed automatically to move data between formats without introducing errors.
+
+# Wide-Names
+
+To generate wide-names using the ODM, there are different formulae depending on the kind of data point you are recording.
+
+## Attributes
```

Comment:
Suggest starting with a short description or reminder that is an attribute, measure or method.